src/js/actions/ui.js

```diff
@@ -99,6 +100,9 @@ define(function (require, exports) {
                 };
 
                 this.dispatch(events.ui.TRANSFORM_UPDATED, payload);
+            })
+            .then(function () {
+                this.flux.actions.tools.resetBorderPolicies();
```

Comment:
I can move it to the previous handler, but if I return this promise, I end up enqueuing resetBorder policies and deadlocking the app.

Maybe this is a bug in our action queue, but it feels like if an action queues another action, it shouldn't wait for it to resolve to consider itself finishing.